python3Packages.qiskit: 0.14.0 -> 0.17.0

drewrisinger opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

pythonPackages.qiskit build was broken because upstream refactored the package into several smaller packages.

This PR fully refactors the Qiskit library into the subpackages, which are all tested as much as possible, and includes new dependency packages as necessary (or updates to existing packages).

###### Things done

My apologies, I know this is a pretty big PR to review. Every new dependency added was necessary for build & tests to pass, minus ``stestr`` which is primarily for faster testing.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`. **No diff detected**
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

**Note: most of these packages are missing maintainers. I'm not sure how that process works, and will update when told.**

drewrisinger committed:
pythonPackages.qiskit: 0.14.0 -> 0.17.0

Updates the qiskit package to follow new Git structure, where
subcomponents are broken into separate packages (terra, aer, etc).

Refactor/cleanup

drewrisinger committed:
pythonPackages.IBMQuantumExperience: remove deprecated package

Package was deprecated, and effectively replaced by
pythonPackages.qiskit-ibmq-provider.

drewrisinger commented:
The failing check seems to be related to removing obsolete IBMQuantumExperience package.

jonringer commented on pkgs/development/libraries/libcint/default.nix:
```diff
@@ -0,0 +1,49 @@
+, cmake
+, openblas
+, python
+, pythonPackages
```
generally you'll want to import only python. Then just access the packages by doing `python.pkgs`

jonringer commented on pkgs/development/libraries/libcint/default.nix:
```diff
@@ -0,0 +1,49 @@
+  ];
+
+  doCheck = true;
+  checkInputs = [ python pythonPackages.numpy ];
```
```suggestion
  checkInputs = [ python.pkgs.numpy ];
```

jonringer replied:
interpreter will be brought in through propagatedBuildInputs

jonringer commented on pkgs/development/libraries/libcint/default.nix:
```diff
@@ -0,0 +1,49 @@
+  checkInputs = [ python pythonPackages.numpy ];
+  checkPhase = ''
+    make test
+  '';
```
I believe this is the default, should be able to remove

jonringer commented on pkgs/development/libraries/libcint/default.nix:
```diff
@@ -0,0 +1,49 @@
+    make test
+  '';
+
+  meta = {
```
```suggestion
  meta = with lib; {
```

jonringer commented on pkgs/development/libraries/libcint/default.nix:
```diff
@@ -0,0 +1,49 @@
+  meta = {
+    description = ''
+      general GTO integrals for quantum chemistry
+    '';
```
I would make one line, and capitallize
```
    description = "General GTO integrals for quantum chemistry";
```

jonringer commented on pkgs/development/libraries/libcint/default.nix:
```diff
@@ -0,0 +1,49 @@
+    homepage = http://wiki.sunqm.net/libcint;
+    downloadPage = https://github.com/sunqm/libcint;
+    license = lib.licenses.bsd2;
+    maintainer = with lib.maintainers; [];
```
please add yourself as a maintainer. 

If you're not already in maintainers.nix, please add yourself in a separate commit, then amend the others. similar to https://github.com/NixOS/nixpkgs/pull/78311

jonringer commented on pkgs/development/libraries/science/math/scs/default.nix:
```diff
@@ -2,13 +2,13 @@
     repo = "scs";
-    rev = "v${version}";
-    sha256 = "17lbcmcsniqlyzgbzmjipfd0rrk25a8hzh7l5wl2wp1iwsd8c3a9";
+    rev = "${version}";
```
```suggestion
    rev = version;
```

jonringer commented on pkgs/development/python-modules/cvxpy/default.nix:
```diff
@@ -0,0 +1,52 @@
+    six
+  ];
+
+  doCheck = true;
```
default for buildPythonPackage
```suggestion
```

jonringer commented on pkgs/development/python-modules/cvxpy/default.nix:
```diff
@@ -0,0 +1,52 @@
+  ];
+
+  doCheck = true;
+  dontUseSetuptoolsCheck = true;
```
you overwrite the default checkPhase, so this isn't useful
```suggestion
```

jonringer commented on pkgs/development/python-modules/cvxpy/default.nix:
```diff
@@ -0,0 +1,52 @@
+  dontUseSetuptoolsCheck = true;
+  checkInputs = [ nose ];
+  checkPhase = ''
+    eval "nosetests"
```
```suggestion
    nosetests
```

jonringer commented on pkgs/development/python-modules/dlx/default.nix:
```diff
@@ -0,0 +1,27 @@
+  meta = {
+    description = ''
+      Implementation of Donald Knuth's Dancing Links algorithm.
+    '';
```
one line please

jonringer commented on pkgs/development/libraries/libcint/default.nix:
```diff
@@ -0,0 +1,49 @@
+      It provides C/Fortran API to evaluate one-electron / two-electron
+      integrals for Cartesian / real-spheric / spinor Gaussian type functions.
+    '';
+    homepage = http://wiki.sunqm.net/libcint;
```
URLs should now be quoted https://github.com/NixOS/rfcs/pull/45, please update the rest
```suggestion
    homepage = "http://wiki.sunqm.net/libcint";
```

jonringer commented on pkgs/development/libraries/subunit/default.nix:
```diff
@@ -1,14 +1,16 @@
+    sha256 = "1hk5v8gg5azwrc079xqigj0sp6qm47vzg1qwjbvb8dbjwlg75rvz";
   };
 
   nativeBuildInputs = [ pkgconfig ];
```
not part of your changes, but the canonical `pkg-config` name is preferred
```suggestion
  nativeBuildInputs = [ pkg-config ];
```

jonringer commented on pkgs/development/python-modules/dlx/default.nix:
```diff
@@ -0,0 +1,27 @@
+  };
+
+  # This package contains no tests
+  doCheck = false;
```
would prefer checking out from github if available. Unit tests are crucial in assuring your package isn't' broken by bumping dependencies

jonringer commented on pkgs/development/python-modules/ecos/default.nix:
```diff
@@ -0,0 +1,50 @@
+  disabled = pythonOlder "3.6";
+
+  # needs ecos submodule, so must be fetchgit vs fetchFromGitHub
+  src = fetchgit {
```
fetchFromGit can also fetchsubmodules, although, under the covers it defaults to using fetchgit

drewrisinger replied:
thanks.

jonringer commented on pkgs/development/python-modules/ecos/default.nix:
```diff
@@ -0,0 +1,50 @@
+    scipy
+  ];
+
+  doCheck = true;
```
```suggestion
```

jonringer commented on pkgs/development/python-modules/ecos/default.nix:
```diff
@@ -0,0 +1,50 @@
+    cp -r ./src/ $out/src # Move tests to $out
+    # Run tests
+    eval "nosetests $out/src/test_interface.py $out/src/test_interface_bb.py"
+  '';
```
should avoid moving unnecessary things to $out
```
    cd src
    # Run tests
    nosetests test_interface.py test_interface_bb.py
```

jonringer commented on pkgs/development/python-modules/fastdtw/default.nix:
```diff
@@ -0,0 +1,36 @@
+      Python implementation of FastDTW.
+      FastDTW is an approximate Dynamic Time Warping (DTW) algorithm that provides
+      optimal or near-optimal alignments with an O(N) time and memory complexity.
+    '';
```
description should be a one line summary. For longer descriptions use `longDescription`

jonringer commented on pkgs/development/python-modules/ibmquantumexperience/default.nix:
```diff
@@ -1,33 +0,0 @@
-
-buildPythonPackage rec {
-  pname = "IBMQuantumExperience";
-  version = "2.0.4";
```
why remove?

drewrisinger replied:
``ibmquantumexperience`` package was deprecated a year ago in favor of ``qiskit-ibmq-provider``, and ``ibmquantumexperience`` is unused in-tree.

https://github.com/QISKit/qiskit-api-py

jonringer commented on pkgs/development/python-modules/marshmallow-polyfield/default.nix:
```diff
@@ -0,0 +1,28 @@
+, fetchPypi
+, lib
+, marshmallow
+}:
```
```suggestion
}:

```

jonringer commented on pkgs/development/python-modules/marshmallow-polyfield/default.nix:
```diff
@@ -0,0 +1,28 @@
+    marshmallow
+  ];
+
+  doCheck = true;
```
default

```suggestion
```

jonringer commented on pkgs/development/python-modules/marshmallow-polyfield/default.nix:
```diff
@@ -0,0 +1,28 @@
+    description = ''
+      An unofficial extension to Marshmallow to allow for polymorphic fields
+    '';
+    license = lib.licenses.asl20;
```
add yourself as maintainer please

jonringer commented on pkgs/development/python-modules/nest-asyncio/default.nix:
```diff
@@ -14,6 +14,11 @@ buildPythonPackage rec {
+  # tests not packaged with source dist as of 1.2.1/1.2.2, and
+  # can't check tests out of GitHub easily without specific commit IDs (no tagged releases)
+  doCheck = false;
+  pythonImportsCheck = [ "nest_asyncio" ];
```
this seems reasonable

jonringer commented on pkgs/development/python-modules/osqp/default.nix:
```diff
@@ -0,0 +1,54 @@
+    future
+  ];
+
+  doCheck = true;
```
```suggestion
```

jonringer commented on pkgs/development/python-modules/osqp/default.nix:
```diff
@@ -0,0 +1,54 @@
+
+  doCheck = true;
+  checkInputs = [ scipy pytestCheckHook mkl ];
+  dontUseSetuptoolsCheck = true;
```
there's no check phase then... please add tests

drewrisinger replied:
``pytestCheckHook`` runs pytest automatically. Setuptools Check doesn't work with several of these packages.

drewrisinger replied:
This line doesn't specifically need it, but it still uses pytestCheckHook. Fixed in commit.

jonringer commented on pkgs/development/python-modules/osqp/default.nix:
```diff
@@ -0,0 +1,54 @@
+    homepage = "https://osqp.org/";
+    downloadPage = "https://github.com/oxfordcontrol/osqp";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [];
```
please add yourself as maintainer

jonringer commented on pkgs/development/python-modules/pproxy/default.nix:
```diff
@@ -0,0 +1,47 @@
+    uvloop
+  ];
+
+  doCheck = true;
```
```suggestion
```

jonringer commented on pkgs/development/python-modules/pproxy/default.nix:
```diff
@@ -0,0 +1,47 @@
+      echo "***Testing $f***"
+      eval "python $f"
+    done
+  '';
```
should be:
```
  checkInputs = [ pytest ];
  checkPhase = ''
    pytest --ignore tests/api_server.py --ignore tests/api_client.py
  '';
```

drewrisinger replied:
I spent about 10 mins trying to get these tests to work with pytest, and I don't think it will work. they're just not designed for a test suite runner at all, and are only meant to be run with ``python $FILE``, due to function naming, requiring command line arguments, and having no ``if __name__ == "__main__"`` guards. Full notes below.

drewrisinger replied:
When I run above, it doesn't collect or run any tests.

When I run ``pytest --ignore[...] ./tests/*.py``, it actually tries to run the tests, but doesn't ignore the proper tests. ``pytest ./tests/cipher_*.py`` works slightly better

Issue is that tests are not guarded by ``if __name__ == "__main__"`` or meant for any test runner (pytest/nose/tox), and that they include code similar to ``cipher = sys.argv[1] if len(sys.argv) > 1 else None``, which grabs ``--ignore=...``

jonringer commented on pkgs/development/python-modules/pproxy/default.nix:
```diff
@@ -0,0 +1,47 @@
+    description = "Proxy server that can tunnel among remote servers by regex rules";
+    homepage = "https://github.com/qwj/python-proxy";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [];
```
add yourself as maintainer please

jonringer commented on pkgs/development/python-modules/pylatexenc/default.nix:
```diff
@@ -0,0 +1,31 @@
+    sha256 = "0wnl00y5dl56aw9j4y21kqapraaravbycwfxdmjsbgl11nk4llx9";
+  };
+
+  doCheck = true;
```
```suggestion
```

jonringer commented on pkgs/development/python-modules/pylatexenc/default.nix:
```diff
@@ -0,0 +1,31 @@
+    description = "Simple LaTeX parser providing latex-to-unicode and unicode-to-latex conversion";
+    downloadPage = "https;//www.github.com/phfaist/pylatexenc";
+    homepage = "https://pylatexenc.readthedocs.io";
+    license     = lib.licenses.mit;
```
add yourself as maintainer please

jonringer commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,70 @@
+
+  PYSCF_INC_DIR = "${libcint}/lib:${libxc}/lib";
+
+  doCheck = true;
```
```suggestion
```

jonringer commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,70 @@
+  meta = {
+    description = ''
+      PySCF: Python-based Simulations of Chemistry Framework
+    '';
```
one line please. Also dont need to say the package name, it's already in context of the package

```
description = "Python-based Simulations of Chemistry Framework";
```

jonringer commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,70 @@
+    homepage = http://www.pyscf.org/;
+    downloadPage = https://github.com/pyscf/pyscf;
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [];
```
add yourself as maintainer please

jonringer commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,70 @@
+      for quantum chemistry calculations and methodology development.
+    '';
+    homepage = http://www.pyscf.org/;
+    downloadPage = https://github.com/pyscf/pyscf;
```
quote URLs

jonringer commented on pkgs/development/python-modules/qiskit-aer/default.nix:
```diff
@@ -0,0 +1,75 @@
+
+    # Add tests & test config to distribution
+    echo "recursive-include test *" >> MANIFEST.in
+    echo "include *.conf" >> MANIFEST.in
```
This shouldn't be necessary, we don't create an sdist

drewrisinger replied:
``scikit-build`` seems to build an sdist then install it. This seemed the easiest way to get the tests into $out.

jonringer commented on pkgs/development/python-modules/qiskit-aer/default.nix:
```diff
@@ -0,0 +1,75 @@
+    "-DBUILD_TESTS=True"
+  ];
+
+  doCheck = true;
```
```suggestion
```

jonringer commented on pkgs/development/python-modules/qiskit-aer/default.nix:
```diff
@@ -0,0 +1,75 @@
+    description = "High performance simulators for Qiskit";
+    homepage = "https://github.com/QISKit/qiskit-aer";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [];
```
please add yourself as maintainer

jonringer commented on pkgs/development/python-modules/qiskit-aqua/default.nix:
```diff
@@ -0,0 +1,79 @@
+  ];
+
+  # Tests take about 12 mins to run on 8-core, parallelized. Not sure if worth running or not.
+  doCheck = true;
```
```suggestion
```

jonringer commented on pkgs/development/python-modules/qiskit-aqua/default.nix:
```diff
@@ -0,0 +1,79 @@
+    scikitlearn
+  ];
+
+  # Tests take about 12 mins to run on 8-core, parallelized. Not sure if worth running or not.
```
unless using something like `pytest-xdist`, these tests will be single threaded anyway, so yes, they are worth. However, if you can conditionally disable the slow ones, that is also acceptable

drewrisinger replied:
stestr theoretically runs them multi-threaded, unless nix does some behind-the-scenes magic not to make that happen.

jonringer commented on pkgs/development/python-modules/qiskit-aqua/default.nix:
```diff
@@ -0,0 +1,79 @@
+    "qiskit.optimization"
+  ];
+  checkPhase = ''
+    cd /build/source && eval "stestr run"
```
```suggestion
    stestr run
```

jonringer commented on pkgs/development/python-modules/qiskit-ibmq-provider/default.nix:
```diff
@@ -0,0 +1,68 @@
+      }
+    )
+  );
+in
```
not allowed in python-modules, will create incoherent imports if another package picks up the default version. If this package just needs to export the executable, then we can move this package out of `python-modules`. But overriding is not allowed in `python-modules`

jonringer commented on pkgs/development/python-modules/qiskit-ignis/default.nix:
```diff
@@ -0,0 +1,66 @@
+
+    cd $out && eval "stestr run"
+
+    runHook postCheck
```
could probably just be
```
  stestr run
```

drewrisinger replied:
Is there a shortcut for getting back to the build source directory (if using fetchFromGitHub: /build/source, but if using fetchPypi: /build/$pname-$version). Most of my issues are because it feels inconsistent where the tests are run from (i.e. what ``pwd`` returns). Sometimes it's $out, sometimes it's /build/source.

jonringer commented on pkgs/development/python-modules/qiskit-ignis/default.nix:
```diff
@@ -0,0 +1,66 @@
+  meta = {
+    description = ''
+      Qiskit tools for quantum hardware verification, noise characterization, and error correction
+    '';
```
Should be able to fit on a single line

jonringer commented on pkgs/development/python-modules/qiskit-terra/default.nix:
```diff
@@ -0,0 +1,102 @@
+    psutil
+    scipy
+    sympy
+  ] ++ (if useOptionals then optionals else []);
```
generally optional dependencies are left up to the user to install.

jonringer commented on pkgs/development/python-modules/qiskit/default.nix:
```diff
@@ -1,65 +1,51 @@
 
-  # Pypi's tarball doesn't contain tests
-  doCheck = false;
+  doCheck = true;
```
```suggestion
```

jonringer commented on pkgs/development/python-modules/scs/default.nix:
```diff
@@ -0,0 +1,60 @@
+
+  prePatch = ''
+    cp -r ${scs.src}/* ./scs
+  '';
```
or
```
  sourceRoot = "source/src";
```

jonringer commented on pkgs/development/python-modules/scs/default.nix:
```diff
@@ -0,0 +1,60 @@
+  doCheck = true;
+  checkInputs = [ nose ];
+  checkPhase = ''
+    eval "nosetests"
```
```suggestion
    nosetests
```

jonringer commented on pkgs/development/python-modules/scs/default.nix:
```diff
@@ -0,0 +1,60 @@
+    scipy
+  ];
+
+  doCheck = true;
```
```suggestion
```

jonringer reviewed (commented):
If you want to split this into many PRs, that would probably be a good idea

jonringer reviewed (commented):
If you want to split this into many PRs, that would probably be a good idea

drewrisinger commented:
> If you want to split this into many PRs, that would probably be a good idea

I can do that. We're just looking at ~20 PRs, which is a pain to submit. Any tips on resolving dependency issues b/c the later PRs won't build w/o the previous PRs being approved?

jonringer commented:
honestly, a package for each PR isn't a terrible idea: limits scope, makes edits easy, faster to review

drewrisinger commented:
Splitting into multiple PRs, one per package. Most issues addressed. My apologies in advance for the deluge of PRs @jonringer

drewrisinger commented:
@jonringer I'd love to get this into 20.03, but my sub-PRs aren't getting reviewed/approved. Is this likely to get pulled in by then? I don't want to be pushy if it's not going to happen, but I also think my PRs are 95%+ ready to be merged. See linked PRs above.

jonringer commented:
there's a few merge conflicts, it would probably take me a full hour to address 27 individual commits. Which I don't have really

drewrisinger commented:
> there's a few merge conflicts, it would probably take me a full hour to address 27 individual commits. Which I don't have really

I can rebase to fix merge conflicts. Is there anyone else I could ask to review/approve to reduce your workload?

drewrisinger commented:
Relevant comments from other threads: https://github.com/NixOS/nixpkgs/pull/78874#pullrequestreview-352646191

https://github.com/NixOS/nixpkgs/pull/78874#issuecomment-581981908

Summary: ``qiskit-aqua`` depends on docplex -> docloud, which is a "smelly" package and unlikely to get approved. I propose limiting scope to just ``qiskit-terra`` (maybe ``qiskit-aer``) FOR NOW, which have fewer dependencies and are likely to get approved faster.

drewrisinger committed:
pythonPackages.ipyvue: init at 1.3.1

drewrisinger committed:
pythonPackages.ipyvuetify: init at 1.2.2

drewrisinger committed:
python3Packages.qiskit-ibmq-provider: 0.5.0 -> 0.6.0

drewrisinger commented:
Ok. After the merge of #83447, I updated this PR with the FINAL STEPS!!!
* updated the ``qiskit-ibmq-provider`` to v0.6.0 (required new jupyter extensions for tests to run properly, I'm fine removing them if people have strong opinions).
* Updated ``python3Packages.qiskit`` to 0.17.0
* Removed deprecated ``python3Packages.IBMQuantumExperience`` (superseded by ``python3Packages.qiskit-ibmq-provider``).

@GrahamcOfBorg build python37Packages.qiskit python38Packages.qiskit
@GrahamcOfBorg eval
(this will probably time out b/c the tests in ``python37Packages.qiskit-aqua`` take > 3600s on ofborg. I can disable a few more of them if preferred). Does anyone know if there are similar timeout issues on Hydra?

drewrisinger committed:
python3Packages.qiskit-aqua: disable slow tests

Arbitrarily disable any tests > 10s on my machine to lighten the build
load (both for borg CI & Hydra builds). All disabled slow tests pass
locally.

drewrisinger commented:
> Yes, please get this <10mins. 

Test run in ~ 5 mins locally. Hope they run in < 10 on borg.
@GrahamcOfBorg build python37Packages.qiskit-aqua
@GrahamcOfBorg build python38Packages.qiskit-aqua
@GrahamcOfBorg build python37Packages.qiskit
@GrahamcOfBorg build python38Packages.qiskit

drewrisinger commented:
@GrahamcOfBorg eval

jonringer commented on pkgs/development/python-modules/ibmquantumexperience/default.nix:
```diff
@@ -1,33 +0,0 @@
-}:
-
-buildPythonPackage rec {
-  pname = "IBMQuantumExperience";
```
please remove associated entry in pkgs/top-level/python-modules.nix

```
getting status of '/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/packet-spot-eval-1/pkgs/development/python-modules/ibmquantumexperience': No such file or directory
```

drewrisinger replied:
ah, whoops. thanks.

drewrisinger commented:
@GrahamcOfBorg build python37Packages.qiskit-aqua
@GrahamcOfBorg build python38Packages.qiskit-aqua
@GrahamcOfBorg build python37Packages.qiskit
@GrahamcOfBorg build python38Packages.qiskit

jonringer commented on pkgs/development/python-modules/ipyvue/default.nix:
```diff
@@ -0,0 +1,32 @@
+    ipywidgets
+  ];
+
+  doCheck = false;  # No tests in package or GitHub
```
verify that it's not trivially broken
```suggestion
  doCheck = false;  # No tests in package or GitHub
  pythonImportsCheck = [ "ipyvue" ];
```

drewrisinger replied:
Thanks, I forgot while doing it quick & dirty. will add.

drewrisinger replied:
done.

jonringer commented on pkgs/development/python-modules/ipyvuetify/default.nix:
```diff
@@ -0,0 +1,31 @@
+    ipyvue
+  ];
+
+  doCheck = false;  # no tests on PyPi/GitHub
```
```suggestion
  doCheck = false;  # no tests on PyPi/GitHub
  pythonImportsCheck = [ "ipyvuetify" ];
```

jonringer reviewed (commented):
otherwise LGTM

jonringer commented on pkgs/development/python-modules/ipyvue/default.nix:
```diff
@@ -0,0 +1,29 @@
+
+buildPythonPackage rec {
+  pname = "ipyvue";
+  version = "1.3.1";
```
```suggestion
  version = "1.3.1";
  disabled = isPy27;
```

```
    File "/nix/store/3n9sxl9nbj5ijn6dnwxajvy3wzy27sdc-python2.7-ipyvue-1.3.1/lib/python2.7/site-packages/ipyvue/__init__.py", line 4, in <module>
      from .VueTemplateWidget import VueTemplate  # noqa: F401
    File "/nix/store/3n9sxl9nbj5ijn6dnwxajvy3wzy27sdc-python2.7-ipyvue-1.3.1/lib/python2.7/site-packages/ipyvue/VueTemplateWidget.py", line 37
      widget = getattr(module, class_name)(**props, model_id=content['id'])
                                                  ^
  SyntaxError: invalid syntax
cannot build derivation '/nix/store/bf7f485yxdh2fcqsiqz923h4cmzilq0a-python2.7-ipyvuetify-1.2.2.drv': 1 dependencies couldn't be built
```

drewrisinger commented:
Ok. Apologies. Built locally with ``nixpkgs-review pr 78772``.
@GrahamcOfBorg eval
Updated in this PR:
@GrahamcOfBorg build python37Packages.qiskit python37Packages.qiskit-ibmq-provider
Added in this PR:
@GrahamcOfBorg build python37Packages.ipyvue python38Packages.ipyvue python37Packages.ipyvuetify python38Packages.ipyvuetify
Dependent of qiskit (enabled now that not broken):
@GrahamcOfBorg build python37Packages.qasm2image

jonringer reviewed (approved):
diff LGTM
commits LGTM

```
https://github.com/NixOS/nixpkgs/pull/78772
9 package built:
python37Packages.ipyvue python37Packages.ipyvuetify python37Packages.qasm2image python37Packages.qiskit python37Packages.qiskit-aqua python37Packages.qiskit-ibmq-provider python38Packages.ipyvue python38Packages.ipyvuetify python38Packages.qiskit-aqua
```

jonringer commented:
it only took two months, but we did it :)

drewrisinger commented:
Yeah, I know. My apologies about the long journey and many messages. Thanks
for your help!

On Tue, Apr 7, 2020, 9:41 PM Jon <notifications@github.com> wrote:

> it only took two months, but we did it :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/78772#issuecomment-610704025>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACNZYI2ZSXUE6H66NNE3HSTRLPI5LANCNFSM4KNFNTDQ>
> .
>
